src/engine/QMLEngine.js

```diff
@@ -493,12 +498,23 @@ class QMLEngine {
 
   // Return a path to load the file
   $resolvePath(file) {
+    if (this.$rootBasePath === undefined || this.$basePath === undefined) {
+      return file;
+    }
+
+    // "Root" references in QML should still be relative to the root QML file
+    if (file.indexOf("/") === 0 && this.$rootBasePath !== undefined) {
```

Comment:
Are you good with this concept of rootBasePath? It requires anything outside of wherever the root QML file is stored to have a full HTTP path. I've started playing around with the qrc system in qmlweb and feel like that might be a better approach to create an "insulated" file structure.